samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,255 @@
+            ],
+        },
+        rf"^arn:{partition_ex}:states:{region_ex}:{account_ex}:stateMachine:[\w-]+$": {
+            "Resource": "AWS::StepFunctions::StateMachine",
+            "Action": [
+                "stepfunction:StartExecution",
+                "stepfunction:StopExecution",
+            ],
+        },
+    }
+    return default_permissions_map
+
+
+def _get_action_mapping(resource_arn: str) -> Union[dict, None]:
+
+    """
+    Returns a dictionary mapping a Resource type to a list of IAM actions generated for that resource.
+
+    Returns `None` if the supplied ARN is for a Resource type without any IAM actions supported.
+
+    Parameters
+    ----------
+    resource_arn:
+        The arn of the resource for which the mapping is returned.
+
+    Returns
+    -------
+    dict:
+        A dictionary containing two keys: `'Resource'` and `'Action'`. `'Resource'` contains the AWS Resource for which the IAM actions list contained in `'Action'` maps to.
+
+        For example `{"Resource": "AWS::S3::Bucket", "Action": ["s3:PutObject", "s3:GetObject"]}`
+
+    """
+
+    permissions_map = _get_permissions_map()
+
+    for arn_exp, mapping in permissions_map.items():
+        if re.search(arn_exp, resource_arn) is not None:
+            LOG.info("Matched ARN `%s` to IAM actions %s", resource_arn, mapping["Action"])
+            return mapping
+
+    # If the supplied ARN is for a resource without any default actions supported
+    return None
+
+
+def _create_iam_statment_string(resource_arn: str) -> str:
+    """
+    Returns an IAM Statement in the form of a YAML string corresponding to the supplied resource ARN.
+
+    The generated Actions are commented out to establish a barrier of consent, preventing customers from accidentally granting permissions that they did not mean to.
+
+    The Action list is empty if there are no IAM Action permissions to generate for the given resource.
+
+    Parameters
+    ----------
+    resource_arn : str
+        The arn of the resource for which the IAM statement is generated.
+
+    Returns
+    -------
+    str:
+        An IAM Statement in the form of a YAML string.
+    """
+
+    # NOTE: The spacing within these strings is needed to keep the resulting YAML indentation valid
+    new_statement_template = (
+        "- Effect: Allow\n"
+        "  Action:\n"
+        "   {%- for action in action_list %}\n"
+        "     # - {{action}}\n"
+        "   {%- endfor %}\n"
+        "  Resource: {{arn}}\n"
+    )
+    mapping = _get_action_mapping(resource_arn)
+
+    if mapping is None:
+        return jinja2.Template(new_statement_template).render(action_list=[], arn=resource_arn).rstrip()
+
+    if mapping["Resource"] in ("AWS::ApiGateway::Api", "AWS::ApiGateway::RestApi"):
+        apiId = _extract_api_id_from_arn(resource_arn)
+        execute_api_arn = f"!Sub arn:${{AWS::Partition}}:execute-api:${{AWS::Region}}:${{AWS::AccountId}}:{apiId}/<STAGE>/GET/<RESOURCE_PATH>"
```

Comment:
They're intended to be filled in by the customer, since there's no way for us to know which resource paths they want to grant. As for the stage, we don't know which stages go with which APIs

Reply:
Can we use wildcards for stage, etc? Will we grant permissions we'd rather let the customer explicitly allow?

Reply:
Alex says it should be reasonable. I think it makes sense, to allow any `GET` request on a given API.

I'll change it :)